tools/dissector/sample_dissector.cpp

```diff
@@ -91,261 +250,523 @@ namespace OpenDDS
       return (next_ == 0) ? 0 : next_->get_link (index - 1);
     }
 
+    void Sample_Field::to_stream(std::stringstream &s, Wireshark_Bundle & p) {
+      size_t location = p.buffer_pos();
+      std::string utf8;
+
+      switch (this->type_id_) {
+      case Char:
+        ACE_CDR::Char char_value;
+        p.serializer >> char_value;
+        s << char_value;
+        break;
+
+      case Boolean:
+        ACE_CDR::Boolean boolean_value;
+        p.serializer >> ACE_InputCDR::to_boolean(boolean_value);
+        if (boolean_value)
+          s << "true";
+        else
+          s << "false";
+        break;
+
+      case Octet:
+        ACE_CDR::Octet octet_value;
+        p.serializer >> ACE_InputCDR::to_octet(octet_value);
+        s << octet_value;
+        break;
+
+      case WChar:
+        ACE_CDR::WChar wchar_value;
+        p.serializer >> ACE_InputCDR::to_wchar(wchar_value);
+        utf16_to_utf8(utf8, &wchar_value);
+        s << utf8;
+        break;
+
+      case Short:
+        ACE_CDR::Short short_value;
+        p.serializer >> short_value;
+        s << short_value;
+        break;
+
+      case Long:
+        ACE_CDR::Long long_value;
+        p.serializer >> long_value;
+        s << long_value;
+        break;
+
+      case LongLong:
+        ACE_CDR::LongLong longlong_value;
+        p.serializer >> longlong_value;
+        s << longlong_value;
+        break;
+
+      case UShort:
+        ACE_CDR::UShort ushort_value;
+        p.serializer >> ushort_value;
+        s << ushort_value;
+        break;
+
+      case ULong:
+        ACE_CDR::ULong ulong_value;
+        p.serializer >> ulong_value;
+        s << ulong_value;
+        break;
+
+      case ULongLong:
+        ACE_CDR::ULongLong ulonglong_value;
+        p.serializer >> ulonglong_value;
+        s << ulonglong_value;
+        break;
+
+      case Float:
+        ACE_CDR::Float float_value;
+        p.serializer >> float_value;
+        s << float_value;
+        break;
+
+      case Double:
+        ACE_CDR::Double double_value;
+        p.serializer >> double_value;
+        s << double_value;
+        break;
+
+      case LongDouble:
+        ACE_CDR::LongDouble longdouble_value;
+        p.serializer >> longdouble_value;
+        s << longdouble_value;
+        break;
+
+      case String:
+        ACE_CDR::Char * string_value;
+        string_value = NULL;
+        p.serializer.read_string(string_value);
+        s << string_value;
+        delete [] string_value;
```

Comment:
I don't think there's any guarantee that the string is allocated with new[].  This block can probably be simplified by using CORBA::String_var or TAO::String_Manager.  Same for WString below.